### PR TITLE
(#380)Translation-Mini-Landing

### LINKED ISSUE #380: CMS: Translate Mini Landing
## As a [role], I want [feature], so that [benefit]

> **ESTIMATE** TBD

### Acceptance Criteria

```gherkin
Scenario: Add a translation
        Given user is navigating to "/user/login"
        When user enters credentials
        And user clicks "Log in" button
        Then user is logged in and the user name "admin" is displayed in the toolbar
        And the tool bar appears at the top
        When user clicks on "Content" tab
        Then user selects "Translate" option from Operations dropdown for content with title "Automated Test Mini Landing Page Edited"
        Then the page title is "Translations of Automated Test Mini Landing Page Edited"
        When user clicks on "Add" button to add translation
        Then page title is "Crear traducción Español de Automated Test Mini Landing Page Edited"
        And the following fields are displayed
            | fieldLabel               | field_name                     |
            | Pretty URL               | field_pretty_url               |
            | Título de página         | title                          |
            | Browser Title            | field_browser_title            |
            | Card Title               | field_card_title               |
            | Meta Description         | field_page_description         |
            | Feature Card Description | field_feature_card_description |
        And "List Description" label is displayed
        And banner image has requirements translated as follows
            | text                                |
            | Máximo 1 fichero.                   |
            | límite de 256 MB.                   |
            | Tipos permitidos: png gif jpg jpeg. |
        And Remove button for image was translated as "Eliminar"
        And the following content sections are displayed
            | section                    |
            | Content Block              |
            | Two Item Feature Card Row  |
            | Listado                    |
            | Borderless Full-Width Card |
            | Raw Html Content           |
            | Dynamic List               |
        And Dynamic List has options translated as "Opciones"
        And button to add content was translated to start with "Añadir"
        And "Search Engine Restrictions" dropdown displays "Include in search"
        And current state was translated as "Estado actual" "Borrador"
        And Change to dropdown has the following options
            | option    |
            | Borrador  |
            | Review    |
            | Publicado |
        And Save button was translated as "Guardar (esta traducción)"
        And preview button was translated as "Vista previa"
        Then user saves the content page
        And user clicks on the tool bar status green button "Borrador"
        And user selects "Quick Publish" from workflow actions

    Scenario: Verify translated content
        Given user is navigating to the front end site with spanish path "/espanol" site section plus "mini-landing-page-edited"
        And page title is "Automated Test Mini Landing Page Edited"
        And Content Heading reads "Content Heading"
        And HTML Content reads "This is HTML content for Content Block."
        And feature card row displays the following cards
            | title                             | url                          | description                       |
            | Article to test Related Resources | {TEST_SITE_SECTION}/article  | N/A                               |
            | External Feature Card             | https://www.externalcard.com | External Feature Card Description |
        And List Item Title is displayed as "List Item Title"
        And managed list has the following links
            | title                                    | url                           | description                                          |
            | Override Test File for Related Resources | {TEST_SITE_SECTION}/article   | Article to test Related Resources - Meta Description |
            | Google Link                              | https://www.google.com        | N/A                                                  |
            | Override Test File for Related Resources | {TEST_SITE_SECTION}/test-file | N/A                                                  |
        And borderless card with "accent-warm" accent displays the following
            | title                             | link                        |
            | Article to test Related Resources | {TEST_SITE_SECTION}/article |
        And the button with text "BUTTON TEXT FOR BORDERLESS FULL-WIDTH CARD" appears with href "{TEST_SITE_SECTION}/article"
        And the Raw HTML Content reads the following "This is to test Raw HTML Content"
        And the Dynamic List Title reads as "Dynamic List Title"
        And 5 links are displayed under the Dynamic List Title and all starts with "/espanol"
        And view title displays text "Press Releases"
        And the page contains meta tags with the following names
            | name        | content                                           |
            | description | Automated Test Mini Landing Page Meta Description |
  
    Scenario: Edit and republish Spansih article content type
        Given user is navigating to "/user/login"
        When user enters credentials
        And user clicks "Log in" button
        Then user is logged in and the user name "admin" is displayed in the toolbar
        And the tool bar appears at the top
        When user clicks on "Content" tab
        And user clicks on title with url spanish path "/espanol" site section plus "mini-landing-page-edited"
        And user clicks on the tool bar status green button "Publicado"
        And user clicks "View in edit form" button from other actions
        And user clears out "Título de página" field
        And user clears out "Meta Description" field
        And user fills out the following fields
            | fieldLabel       | value                                                  | field_name             |
            | Título de página | Automated Test Mini Landing Edited Spanish                  | title                  |
            | Meta Description | Automated Test Mini Landing Meta Description Edited Spanish | field_page_description |
        When user saves the content page
        And user clicks on the tool bar status green button "Editing"
        And user selects "Quick Publish" from workflow actions

    Scenario: Verify Spanish edited content
        Given user is navigating to the front end site with spanish path "/espanol" site section plus "mini-landing-page-edited"
        And browser waits
        Then page title is "Automated Test Mini Landing Edited Spanish "
        And the page contains meta tags with the following names
            | name        | content                                                |
            | description | Automated Test Mini Landing Meta Description Edited Spanish|
```

### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* Web Page: *Here goes a URL to an existing web page*

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Short Task Description - Issue #9999
- [ ] Short Task Description - Issue #9999

### Technical Notes
- More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by ahmadakbaryNIH**
closes #380

### CODE REVIEW

**Comment by ahmadakbaryNIH** [comment]
image

File: cypress/e2e/CMS/$MiniLandingPageTest.feature [hunk 1]
```diff
@@ -251,6 +245,110 @@ Feature: As a cms user I want to be able to create Mini Landing page content typ
...
+        And the button with text "BUTTON TEXT FOR BORDERLESS FULL-WIDTH CARD" appears with href "{TEST_SITE_SECTION}/article"
+        And the Raw HTML Content reads the following "This is to test Raw HTML Content"
+        And the Dynamic List Title reads as "Dynamic List Title"
+        And 6 links are displayed under the Dynamic List Title and all starts with espanol
+        And view title displays text "Press Releases"
+        And the page contains meta tags with the following names
+            | name        | content                                                  |
+            | description | Automated Test Mini Landing Page Meta Description Edited |
+
+    Scenario: Edit and republish Spansih article content type
```
Comment (KateMashkinaNIH) [inline cypress/e2e/CMS/$MiniLandingPageTest.feature:325]: can you rename this scenario as Edit and republish Spanish Mini landing page

File: cypress/e2e/CMS/$MiniLandingPageTest/index.js [hunk 2]
```diff
@@ -232,4 +232,32 @@ Then('the promo image is matching the earlier selected image', () => {
...
+And('button to add content was translated to start with {string}', (TranslatedOption) => {
+    cy.get("input[value='Añadir Content Block']").contains(TranslatedOption)
+})
+
+Given('user is navigating to the front end site with spanish path {string} site section plus {string}', (spPath, purl) => {
+    cy.visit(`${frontEndBaseUrl}${spPath}${siteSection}/${purl}-${randomNum}`, { retryOnStatusCodeFailure: true });
+});
+
+And('{int} links are displayed under the Dynamic List Title and all starts with espanol', (linksNum) => {
+    cy.get("div[class*='dynamic list'] a").should('be.visible').and('have.length', linksNum)
```
Comment (KateMashkinaNIH) [inline cypress/e2e/CMS/$MiniLandingPageTest/index.js:262]: this step does not check if the links start with "/espanol"

Also, the original AC was written as   And 5 links are displayed under the Dynamic List Title and all starts with "/espanol"
Why did you change it without letting me know? "/espanol" is passed as a parameter, you can't just remove it , its part of Acceptance Criteria of a ticket

File: cypress/e2e/CMS/$MiniLandingPageTest/index.js [hunk 3]
```diff
@@ -232,4 +232,32 @@ Then('the promo image is matching the earlier selected image', () => {
...
+        cy.get("div[class*='paragraph-type-ti']").should('include.text', section)
+    }
+})
+
+And('Dynamic List has options translated as {string}', (TranslatedOption) => {
+    cy.get(`tbody summary[role='button'] span:contains("${TranslatedOption}")`).should('be.visible')
+})
+
+And('button to add content was translated to start with {string}', (TranslatedOption) => {
+    cy.get("input[value='Añadir Content Block']").contains(TranslatedOption)
```
Comment (KateMashkinaNIH) [inline cypress/e2e/CMS/$MiniLandingPageTest/index.js:254]: Anadir should not be hardcoded , also I do not see a assertion here 

cy.get("input[value*='Content Block']").should("include.text",TranslatedOption)

**Review by KateMashkinaNIH — CHANGES_REQUESTED** [review]
